pkg/catalog/outbound_traffic_policies.go

```diff
@@ -201,3 +202,92 @@ func (mc *MeshCatalog) getDestinationServicesFromTrafficTarget(t *access.Traffic
 	}
 	return destServices, nil
 }
+
+// GetWeightedClustersForUpstream returns Envoy cluster weights for the given
+// upstream service, the apex service of a TrafficSplit.
+func (mc *MeshCatalog) GetWeightedClustersForUpstream(upstream service.MeshService) []*xds_tcp_proxy.TcpProxy_WeightedCluster_ClusterWeight {
```

Comment:
The return type here seems weird to me. The function returns a weighted cluster for an upstream regardless of its protocol, but we are returning a XDS TcpProxy object here. Could you use `service.WeightedCluster` instead?

We have been intentional in not exposing XDS structs in the catalog interface so far. Has this changed?

Reply:
Makes sense. Just fixed.